src/main/java/picocli/CommandLine.java

```diff
@@ -5603,8 +5593,8 @@ static Range adjustForType(Range result, IAnnotatedElement member) {
             return result.isUnspecified ? defaultArity(member) : result;
         }
         /** Returns the default arity {@code Range}: for interactive options/positional parameters,
-         * this is 0; for {@link Option options} this is 0 for booleans and 1 for
-         * other types, for {@link Parameters parameters} booleans have arity 0, arrays or Collections have
+         * this is 0; for {@link Option options} this is "0..1" for booleans and 1 for
+         * other types, for {@link Parameters parameters} booleans have arity 1, arrays or Collections have
```

Comment:
According to this table https://picocli.info/#_default_arity. For boolean parameters it should be 1. Which one is correct @remkop?

Reply:
You are correct: for boolean positional parameters, the `defaultArity` method returns `Range.valueOf("1").unspecified(true)`.

For named boolean options, the story is a bit more complex.
Strictly speaking, the `defaultArity` method returns `Range.valueOf("0").unspecified(true)`.  _Effectively_, that is "0..1". Shall we change this javadoc to something like this?

> (...); for {@link Option options} this is "0" for booleans (but effectively "0..1" since {@code unspecified} is set to {@code true}) and 1 for other types, for {@link Parameters parameters} booleans have arity 1 (...)

Reply:
That seems quite a bit of implementation details, not sure it should be in the Javadoc. I'm not sure what would be the best way of phrasing that to put in the documentation. Any suggestions? Feel free to change the MR at will.

Reply:
OK, another idea is to use your phrasing, with one small change, add the word _effectively_:

> this is effectively `"0..1"` for booleans

And then we can add an Implementation Notes section to the Javadoc for that method, like this:

```
<p><b><em>Implementation Notes</em></b></p>
<p>The returned {@code Range} for boolean options has an <em>effective</em> arity of "0..1".
This is implemented by returning a {@code Range} with arity "0", 
and its {@code unspecified} property set to {@code true}. 
This implementation may change in the future.</p>
```

Thoughts?

Reply:
Great, thank you!